code/datums/gamemodes/war.dm

```diff
@@ -0,0 +1,76 @@
+#define KOSMNAZ_VICTORY "KOSMNAZ_VICTORY"
+#define NATSF_VICTORY "NATSF_VICTORY"
+
+//War, an HvH mode where players must kill or capture specific areas in order to drain the opposing faction's tickets. - Tennessee116
+
+/datum/game_mode/war
+	name = "War"
+	config_tag = "War"
+	required_players = 0
+	votable = TRUE
+	var/natsf_tickets = 100
+	var/kosmnaz_tickets = 100
+
+/datum/game_mode/war/announce()
+	to_chat(world, "<b>The current game mode is War!</b>")
+	to_chat(world, "<b>Assume DEFCON 1, prepare for all-out war!</b>")
+
+/datum/game_mode/war/proc/kill_tickets(var/mob/living/carbon/human/H)
+	if(H.undefibbable == 1)
+		if(H.faction == "NATSF")
+			natsf_tickets -= 100
+		if(H.faction == "KOSMNAZ")
+			kosmnaz_tickets -= 100
+	return
+
+/datum/game_mode/war/proc/cap_tickets(var/obj/machinery/capbeacon/B)
+	if(B.controlled_by == null)
+		return
+	if(B.controlled_by == "NATSF")
+		kosmnaz_tickets -= 100
+	if(B.controlled_by == "KOSMNAZ")
+		natsf_tickets -= 100
+
+/datum/game_mode/war/proc/ticket_processing()
+	. = ..()
+	kill_tickets(loc()
+	cap_tickets()
+	if(natsf_tickets <= 0)

```

Comment:
this shit is runtiming, I sorta tried fixing it, failed, and it was left with a random loc